ArduSub/control_althold.cpp

```diff
@@ -49,7 +49,30 @@ void Sub::althold_run()
 
     // get pilot desired lean angles
     float target_roll, target_pitch;
-    get_pilot_desired_lean_angles(channel_roll->get_control_in(), channel_pitch->get_control_in(), target_roll, target_pitch, attitude_control.get_althold_lean_angle_max());
+
+    // Check if set_attitude_target_no_gps is valid
+    if (tnow - sub.set_attitude_target_no_gps.status.last_message_ms < 5000 && sub.set_attitude_target_no_gps.status.is_valid) {
+        float target_yaw;
+        Quaternion(
+            set_attitude_target_no_gps.packet.q
+        ).to_euler(
+            target_roll,
+            target_pitch,
+            target_yaw
+        );
+        target_roll = degrees(target_roll);
+        target_pitch = degrees(target_pitch);
+        target_yaw = degrees(target_yaw);
+
+        attitude_control.input_euler_angle_roll_pitch_yaw(target_roll * 1e2, target_pitch * 1e2, target_yaw * 1e2, true);
+        return;
+
+    } else {
+        // Set valid to false, avoiding overflow in last_message_ms
+        sub.set_attitude_target_no_gps.status.is_valid = false;
```

Comment:
I do not see why it is necessary to have the is_valid flag. We can just check the timestamp.